indexprovider/wrapper.go

```diff
@@ -551,69 +559,88 @@ func (w *Wrapper) MultihashLister(ctx context.Context, prov peer.ID, contextID [
-		// database. Pause ingestion until the user can fix the DB.
-		e := fmt.Errorf("getting deal with proposal cid %s from boost database: %w", proposalCid, boostErr)
-		log.Infow("pausing ingestion", "proposalCid", proposalCid, "err", e)
-		return nil, e
-	}
+		// Deal was not found in boost DB - check in legacy markets
+		md, legacyErr := w.legacyProv.GetLocalDeal(proposalCid)
+		if legacyErr == nil {
+			// Found the deal, get an interator over the piece
+			return provideF(proposalCid.String(), false, md.Proposal.PieceCID)
+		}
+
+		// Check if it's a "not found" error
+		if !errors.Is(legacyErr, datastore.ErrNotFound) {
+			// It's not a "not found" error: there was a problem accessing the
+			// legacy database. Pause ingestion until the user can fix the legacy DB.
+			e := fmt.Errorf("getting deal with proposal cid %s from Legacy Markets: %w", proposalCid, legacyErr)
+			log.Infow("pausing ingestion", "proposalCid", proposalCid, "err", e)
+			return nil, e
+		}
 
-	// Deal was not found in boost DB - check in legacy markets
-	md, legacyErr := w.legacyProv.GetLocalDeal(proposalCid)
-	if legacyErr == nil {
-		// Found the deal, get an interator over the piece
-		return provideF(proposalCid, md.Proposal.PieceCID)
+		// The deal was not found in the boost or legacy database.
+		// Skip this deal and continue ingestion.
+		err = fmt.Errorf("deal with proposal cid %s not found", proposalCid)
+		log.Infow("skipping ingestion", "proposalCid", proposalCid, "err", err)
+		return nil, skipError(err)
 	}
 
-	// Check if it's a "not found" error
-	if !errors.Is(legacyErr, datastore.ErrNotFound) {
-		// It's not a "not found" error: there was a problem accessing the
-		// legacy database. Pause ingestion until the user can fix the legacy DB.
-		e := fmt.Errorf("getting deal with proposal cid %s from Legacy Markets: %w", proposalCid, legacyErr)
-		log.Infow("pausing ingestion", "proposalCid", proposalCid, "err", e)
-		return nil, e
+	dealUUID, err := uuid.FromBytes(contextID)
+	if err == nil {
+		// Look up deal by dealUUID in the direct deals database
+		entry, dderr := w.directDealsDB.ByID(ctx, dealUUID)
+		if dderr == nil {
+			// Found the deal, get an iterator over the piece
+			return provideF(dealUUID.String(), true, entry.PieceCID)
+		}
+
+		// Check if it's a "not found" error
+		if !errors.Is(dderr, sql.ErrNoRows) {
+			// It's not a "not found" error: there was a problem accessing the
+			// database. Pause ingestion until the user can fix the DB.
+			e := fmt.Errorf("getting deal with UUID %s from direct deal database: %w", dealUUID, dderr)
+			log.Infow("pausing ingestion", "deal UUID", dealUUID, "err", e)
+			return nil, e
+		}
 	}
 
-	// The deal was not found in the boost or legacy database.
-	// Skip this deal and continue ingestion.
-	err = fmt.Errorf("deal with proposal cid %s not found", proposalCid)
-	log.Infow("skipping ingestion", "proposalCid", proposalCid, "err", err)
```

Comment:
I think we still want to preserve this "not found" error message in the case that the context ID is a uuid but it's not found in the boost DB